lib/release_manager.rb

```diff
@@ -10,8 +10,10 @@ module ReleaseManager
   AGENT_URL      = 'git@github.com:puppetlabs/puppet-agent.git'
   COMPONENTS_DIR = RELEASE_DIR.join('pkg')
   AGENT_DIR      = RELEASE_DIR.join('puppet-agent')
+  LOG_DIR        = ROOT_DIR.join('log')
 
   def self.logger
+    Helpers::File.create_dir(LOG_DIR) unless Helpers::File.dir_exists?(LOG_DIR)
```

Comment:
Since the @logger is a kind of a singleton in this case, change this so that `Helpers::File.dir_exists?(LOG_DIR)` is not evaluated in every method call.